cxx/pclean/schema_helper_test.cc

```diff
@@ -54,22 +54,43 @@ BOOST_AUTO_TEST_CASE(test_get_class_by_name) {
   BOOST_TEST(c.name == "Practice");
 }
 
-BOOST_AUTO_TEST_CASE(test_get_parent_classes) {
+BOOST_AUTO_TEST_CASE(test_domains_cache) {
   PCleanSchemaHelper schema_helper(schema);
-  BOOST_TEST(schema_helper.get_parent_classes("School").empty());
-  BOOST_TEST(schema_helper.get_parent_classes("City").empty());
-  BOOST_TEST(schema_helper.get_parent_classes("Physician").size() == 1);
-  BOOST_TEST(schema_helper.get_parent_classes("Practice").size() == 1);
-  BOOST_TEST(schema_helper.get_parent_classes("Record").size() == 2);
+
+  std::vector<std::string> expected_domains = {"School"};
+  BOOST_TEST(schema_helper.domains["School"] == expected_domains);
+
+  expected_domains = {"Physician", "school:School"};
+  BOOST_TEST(schema_helper.domains["Physician"] == expected_domains);
+
+  expected_domains = {"City"};
+  BOOST_TEST(schema_helper.domains["City"] == expected_domains);
+
+  expected_domains = {"Practice", "city:City"};
+  BOOST_TEST(schema_helper.domains["Practice"] == expected_domains);
+
+  expected_domains = {
+    "Record", "physician:Physician", "physician:school:School",
```

Comment:
How do you plan to indicate that `School`, `school:School`, `physician:school:School` refer to the same domain? In the current HIRM code, domains with different string-names are assumed to be different domains, and we need these to be the same domain so that the clustering does the right thing. Would it work to just use `School` for all of these?

Also, the current code assumes that if a base relation has `k` domains and a noisy relation has `n  > k` domains, the first `k` domains of the noisy relation are the same domains/in the same order as the base relation domains. (So we'll either want to enforce that here, or find another way to associate base/noisy relation domains.)

Reply:
Excellent points!  Done.